config/initializers/sidekiq.rb

```diff
@@ -9,18 +9,20 @@
   }
   Sidekiq::ReliableFetch.setup_reliable_fetch!(config)
 
-  config.server_middleware do |chain|
-    require 'prometheus_exporter/instrumentation'
-    chain.add PrometheusExporter::Instrumentation::Sidekiq
-  end
-  config.on :startup do
-    require 'prometheus_exporter/instrumentation'
-    PrometheusExporter::Instrumentation::Process.start type: 'sidekiq'
-  end
-  config.death_handlers << PrometheusExporter::Instrumentation::Sidekiq.death_handler
+  unless Rails.env.test?
```

Comment:
In development, metrics can be collected just fine because each pod spins up a server. You shouldn't face any issues at all, even if developing directly outside of containers. In test, the rake task doesn't spin up a metrics server, so we can't send metrics anywhere.